.github/workflows/build_and_test_component.yml

```diff
@@ -0,0 +1,56 @@
+run-name: Build and Test Component - Build ${{ inputs.component }}
+name:  Build and Test Component
+
+on:
+  workflow_dispatch:
+    inputs:
+      source_reference:
+        description: |
+          Branch from wazuh/wazuh-agent repository to use.
+        required: true
+      component:
+        description: 'Default agent'
+        required: true
+        type: choice
+          - agent
+        default: agent
+
+  workflow_call:
+    inputs:
+      source_reference:
+        type: string
+        required: true
+      components:
+        type: string
+        required: true
+
+jobs:
+  Build-component-and-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    name: Build ${{ inputs.components }}
+
+    steps:
+      - name: Checkout wazuh/wazuh-agent repository
+        uses: actions/checkout@v4
+        with:
+          repository: wazuh/wazuh-agent
+          ref: ${{ inputs.source_reference }}
+
+      - name: Initializing Submodules
```

Comment:
This step could be removed since the previous one can be set to fetch the `wazuh-agent` repository with its submodules given a certain `ref`. An example can be seen here:
* https://github.com/wazuh/wazuh-agent/blob/524768a0ca71c1ecdd52d742f3923842a672b118/.github/workflows/compile_and_run_tests.yml#L20